dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java

```diff
@@ -329,6 +334,79 @@ public void Test_Non_Admin_User_With_Portlet_Read_Attempt() throws DotDataExcept
         api.saveSecrets(builder.build(), APILocator.systemHost() , nonAdminUserWithAccessToPortlet);
     }
 
+    @Test
+    public void Test_BytesToChars_No_Middle_String_Conversion() throws IOException {
```

Comment:
all the tests here can be implemented as tests cases in a data provider to make the code cleaner